### PR TITLE
Simplify simple blocks by using Block-subclasses equivalent to Vanilla, allowing Mining Gadgets etc. to interact with them

### LINKED ISSUE #37: 1) Change simple blocks to not extend AbstractBlock (which implements EntityBlock) and 2) Make ore blocks drop XP if not already the case

### PR DESCRIPTION
**Pull request by shartte**
Fixes #37

### CODE REVIEW

File: src/main/java/owmii/powah/lib/block/AbstractBlock.java [hunk 1]
```diff
@@ -41,7 +41,7 @@
 import owmii.powah.lib.registry.IVariant;
 import owmii.powah.lib.registry.IVariantEntry;
 
-public class AbstractBlock<V extends IVariant, B extends AbstractBlock<V, B>> extends Block implements IVariantEntry<V, B>, IBlock<V, B> {
+public abstract class AbstractBlock<V extends IVariant, B extends AbstractBlock<V, B>> extends Block implements IVariantEntry<V, B>, IBlock<V, B> {
```
Comment (Technici4n) [inline src/main/java/owmii/powah/lib/block/AbstractBlock.java:44]: funny change :D

**Review by Technici4n — APPROVED** [review]

**Comment by Technici4n** [comment]
Fixes part of https://github.com/Technici4n/Powah/issues/37.

File: src/main/java/owmii/powah/lib/block/AbstractBlock.java (hunk 1 above)
Comment (shartte) [reply src/main/java/owmii/powah/lib/block/AbstractBlock.java:44]: Should the ore drop XP? I just copy pasted iron ore, to be honest.

Comment (Technici4n) [reply src/main/java/owmii/powah/lib/block/AbstractBlock.java:44]: Probably not if iron doesn't. I don't really care

**Pull request merged**